common/Extensions/ObservableCollectionExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace DevHome.Common.Extensions;
+
+public static class ObservableCollectionExtensions
+{
+    public static void Add<T>(this ObservableCollection<T> collection, T item, Func<T, IComparable> keySelector)
```

Comment:
Is this more like an Insert? I'm wondering if it would be useful to have a more specific name, or no and it's totally obvious from the parameters what is happening here.

Reply:
I was thinking about Insert too, but usually we pass a position with an Insert call, so didn't call it Insert. How about SortedAdd?

Reply:
Is this method for inserting an item while maintaining a sorted list?
If yes, I do have a few suggestion and concern:
Concern:
- An `ObservableCollection` will notify subscribers on each insertion and removal. In this case, when clearing and then inserting we may be triggering more events than we want to reflect on the UI.
  - Reference: [observablecollection.cs](https://referencesource.microsoft.com/#System/compmod/system/collections/objectmodel/observablecollection.cs,210)
 
Suggestion:
- I was going to suggest `AddSorted` method but that assumes that the user will only be calling `AddSorted` at all time when inserting into the collection, otherwise the order is not maintained. I am thinking `AddSorted` may not be an extension candidate, instead a sorted collection could potentially be a data structure that ensures the collection is sorted at any point when inserting or removing items.
- If implementing and maintaining a sorted collection is an overkill, I suggest moving this code next to the caller since it will likely be a challenge to easily integrate it in other places.

Reply:
What about making a new class? `SortedObservableCollection`?  The backing collection is an auto-sorted collection and the `Get` returns an `ObservableCollection`?